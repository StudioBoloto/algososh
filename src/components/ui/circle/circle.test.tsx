import React from 'react';
import { render } from '@testing-library/react';
import { Circle } from './circle';
import {ElementStates} from "../../../types/element-states";

describe('Circle Component', () => {
    it('renders correctly without letter', () => {
        const { container } = render(<Circle />);
        expect(container).toMatchSnapshot();
    });

    it('renders correctly with letter', () => {
        const { container } = render(<Circle letter="A" />);
        expect(container).toMatchSnapshot();
    });

    it('renders correctly with head', () => {
        const { container } = render(<Circle head="Head" />);
        expect(container).toMatchSnapshot();
    });

    it('renders correctly with React element in head', () => {
        const { container } = render(<Circle head={<span>Head Element</span>} />);
        expect(container).toMatchSnapshot();
    });

    it('renders correctly with tail', () => {
        const { container } = render(<Circle tail="Tail" />);
        expect(container).toMatchSnapshot();
    });

    it('renders correctly with React element in tail', () => {
        const { container } = render(<Circle tail={<span>Tail Element</span>} />);
        expect(container).toMatchSnapshot();
    });

    it('renders correctly with index', () => {
        const { container } = render(<Circle index={1} />);
        expect(container).toMatchSnapshot();
    });

    it('renders correctly with isSmall prop', () => {
        const { container } = render(<Circle isSmall />);
        expect(container).toMatchSnapshot();
    });

    it('renders correctly in default state', () => {
        const { container } = render(<Circle state={ElementStates.Default} />);
        expect(container).toMatchSnapshot();
    });

    it('renders correctly in changing state', () => {
        const { container } = render(<Circle state={ElementStates.Changing} />);
        expect(container).toMatchSnapshot();
    });

    it('renders correctly in modified state', () => {
        const { container } = render(<Circle state={ElementStates.Modified} />);
        expect(container).toMatchSnapshot();
    });
});
