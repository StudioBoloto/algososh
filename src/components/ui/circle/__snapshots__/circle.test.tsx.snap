// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Component renders correctly in changing state 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle Component renders correctly in default state 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle Component renders correctly in modified state 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle Component renders correctly with React element in head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <span>
        Head Element
      </span>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle Component renders correctly with React element in tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <span>
        Tail Element
      </span>
    </div>
  </div>
</div>
`;

exports[`Circle Component renders correctly with head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      Head
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle Component renders correctly with index 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      1
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</div>
`;

exports[`Circle Component renders correctly with isSmall prop 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle Component renders correctly with letter 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        A
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle Component renders correctly with tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      Tail
    </div>
  </div>
</div>
`;

exports[`Circle Component renders correctly without letter 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;
